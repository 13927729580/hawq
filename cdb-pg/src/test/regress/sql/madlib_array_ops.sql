SET client_min_messages TO ERROR;
DROP SCHEMA IF EXISTS madlib_install_check_gpsql_array_ops CASCADE;
CREATE SCHEMA madlib_install_check_gpsql_array_ops;
SET search_path = madlib_install_check_gpsql_array_ops, madlib;

SELECT
    assert(abs(res1 + res2 + res3 - 17371.7254700712) < 0.01, 'FAIL')
FROM
(
    SELECT
        madlib.array_dot(
            madlib.array_mult(madlib.array_add(an,b),
            madlib.array_sub(an,b)),
        madlib.array_mult(
            madlib.array_div(an,b),
            madlib.normalize(an))) res1 
    FROM (SELECT array[1,2,3]::FLOAT[] an, array[4,5,7]::FLOAT[] b) t
) t1,
(
    SELECT
        (madlib.array_max(b) + 
         madlib.array_min(b) + 
         madlib.array_sum(b) +
         madlib.array_sum_big(b) +
         madlib.array_mean(b) + 
         madlib.array_stddev(b)) res2 
    FROM (SELECT array[4,5,7,NULL]::FLOAT[] b) t
) t2,
(
    SELECT
        madlib.array_sum(
            madlib.array_scalar_mult(
                madlib.array_fill(
                    madlib.array_of_float(20),
                    234.343::FLOAT8),3.7::FLOAT)) res3
) t3;

CREATE table test AS SELECT generate_series(1,100) x;
SELECT array(SELECT unnest(madlib.array_agg(x)) order by 1) FROM test;

DROP SCHEMA madlib_install_check_gpsql_array_ops CASCADE;
