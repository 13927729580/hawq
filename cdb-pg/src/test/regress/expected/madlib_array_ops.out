SET client_min_messages TO ERROR;
DROP SCHEMA IF EXISTS madlib_install_check_gpsql_array_ops CASCADE;
CREATE SCHEMA madlib_install_check_gpsql_array_ops;
SET search_path = madlib_install_check_gpsql_array_ops, madlib;
SELECT
    assert(abs(res1 + res2 + res3 - 17371.7254700712) < 0.01, 'FAIL')
FROM
(
    SELECT
        madlib.array_dot(
            madlib.array_mult(madlib.array_add(an,b),
            madlib.array_sub(an,b)),
        madlib.array_mult(
            madlib.array_div(an,b),
            madlib.normalize(an))) res1 
    FROM (SELECT array[1,2,3]::FLOAT[] an, array[4,5,7]::FLOAT[] b) t
) t1,
(
    SELECT
        (madlib.array_max(b) + 
         madlib.array_min(b) + 
         madlib.array_sum(b) +
         madlib.array_sum_big(b) +
         madlib.array_mean(b) + 
         madlib.array_stddev(b)) res2 
    FROM (SELECT array[4,5,7,NULL]::FLOAT[] b) t
) t2,
(
    SELECT
        madlib.array_sum(
            madlib.array_scalar_mult(
                madlib.array_fill(
                    madlib.array_of_float(20),
                    234.343::FLOAT8),3.7::FLOAT)) res3
) t3;
 assert 
--------
 
(1 row)

CREATE table test AS SELECT generate_series(1,100) x;
SELECT array(SELECT unnest(madlib.array_agg(x)) order by 1) FROM test;
                                                                                                                                               ?column?                                                                                                                                                
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100}
(1 row)

DROP SCHEMA madlib_install_check_gpsql_array_ops CASCADE;
